sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1517,6 +1521,142 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Gets the application details belonging to an institution location.
+   * @param locationId location id.
+   * @param applicationId application id.
+   * @returns student application.
+   */
+  async getApplicationInfo(
+    locationId: number,
+    applicationId: number,
+  ): Promise<Application> {
+    return this.repo.findOne({
+      select: {
+        id: true,
+        data: true as unknown,
+        programYear: {
+          id: true,
+          active: true,
+          startDate: true,
+          endDate: true,
+        },
+        student: {
+          id: true,
+          birthDate: true,
+          user: {
+            id: true,
+            lastName: true,
+          },
+          sinValidation: {
+            id: true,
+            sin: true,
+          },
+        },
+        currentAssessment: {
+          id: true,
+          offering: {
+            id: true,
+            offeringIntensity: true,
+          },
+        },
+      },
+      relations: {
+        programYear: true,
+        location: true,
+        student: {
+          user: true,
+          sinValidation: true,
+        },
+        currentAssessment: {
+          offering: true,
+        },
+      },
+      where: {
+        id: applicationId,
+        location: {
+          id: locationId,
+        },
+        applicationStatus: Not(ApplicationStatus.Overwritten),
+      },
+    });
+  }
+
+  /**
+   * Set of validations for existing application and newly
+   * selected offering, like, application belongs to the location,
+   * newly selected offering belongs to the location, is there any
+   * overlapping study dates for the student, is the offering is
+   * matching with the existing offering and the newly selected
+   * offering belong to the same program year.
+   * @param applicationId existing application id.
+   * @param selectedOffering newly selected offering id.
+   * @param locationId location id.
+   * @param options more options of the validation.
+   * - `isPir` set true, if its PIR related.
+   */
+  async applicationOfferingValidation(
+    applicationId: number,
+    selectedOffering: number,
+    locationId: number,
+    options?: {
+      isPir: boolean;
+    },
+  ): Promise<void> {
+    // Validate if the application exists and the location has access to it.
+    const application = options?.isPir
+      ? await this.getProgramInfoRequest(locationId, applicationId)
```

Comment:
I was under the impression that we are going to retrieve the result required for performing the common validations here. I think we are doing the non-centralized part of validation also here based on option.(e.g. Only PIR specific).

was thinking if we should keep `applicationOfferingValidation` exclusively doing common validations or being versatile like the way it is right now? @andrewsignori-aot @guru-aot @sh16011993 thoughts?

Reply:
On the same topic, I believe that we can have the `application` retrieve by only one method. I am not able to easily find the reason why we need two different methods. Can you please elaborate or can we have a sync up on this?

Reply:
@dheepak-aot as per my understanding, since the method is centralized and no longer has different behavior for PIR and non-PIR, I believe that we achieved the needed centralization, right? 
It looks good to me, at least, unless I am missing something else.